src/openrct2/ride/Vehicle.cpp

```diff
@@ -7460,14 +7460,15 @@ static void vehicle_update_scenery_door(rct_vehicle* vehicle)
         return;
     }
 
-    if (vehicle->next_vehicle_on_train != SPRITE_INDEX_NULL)
+    if ((vehicle->next_vehicle_on_train != SPRITE_INDEX_NULL) && (tileElement->GetAnimationFrame() == '\0'))
```

Comment:
Why compare a frame number to a character? It's just a `uint8_t`.

Reply:
Well I'm questioning my sanity now, because yesterday when I was looking at the definitions of  GetAnimationFrame() and SetAnimationFrame(), I saw that it was taking a char and bitshifting it. 

But now I'm poking around in Visual Studio and that's clearly not the case. Anyway, I'll fix it and test it again.

Reply:
Appveyor failed on an unrelated file:

```ERROR[c:\projects\openrct2\src\openrct2\world\map.cpp:188 (map_get_first_element_at)]: Trying to access element outside of range```

Otherwise I think it's good to go.

Reply:
That is unrelated. It is also an error printed by OpenRCT2 itself. (Not likely to be caused by your branch either.)